src/main/java/com/avispl/symphony/dal/communicator/barco/clickshare/BarcoClickShareCommunicator.java

```diff
@@ -190,23 +294,353 @@ protected void authenticate() throws Exception {
+        }
+
+        v2RequestPowerStatus(statistics, controls);
+        v2RequestVideoMode(statistics, controls);
+        v2RequestAudioMode(statistics, controls);
+        v2RequestNetworkConfiguration(statistics);
+        v2RequestDeviceIdentity(statistics);
+        v2RequestFeatures(statistics, controls);
+        v2RequestPersonalizationFeatures(statistics, controls);
+    }
+
     @Override
     protected HttpHeaders putExtraRequestHeaders(HttpMethod httpMethod, String uri, HttpHeaders headers) throws Exception {
-        if(httpMethod.equals(HttpMethod.PATCH)){
+        if(httpMethod.equals(HttpMethod.PATCH) || httpMethod.equals(HttpMethod.PUT)){
             headers.set(HttpHeaders.CONTENT_TYPE, ContentType.APPLICATION_FORM_URLENCODED.getMimeType());
         }
         headers.set("Authorization", "Basic " + encodedCreds);
         return super.putExtraRequestHeaders(httpMethod, uri, headers);
     }
 
+    /**
+     * Collecting statistics and adding controls for API V1.0
+     * @param statistics map to store statistics data to
+     * @param controls list to add controllable properties to
+     * @throws Exception during http communication
+     */
+    private void v1_0_RequestDeviceInfo(Map<String, String> statistics, List<AdvancedControllableProperty> controls) throws Exception {
+        JsonNode deviceInfo = getApiV1JsonNode(supportedApiVersion + V1_DEVICE_INFO);
+        JsonNode audioEnabled = getApiV1JsonNode(supportedApiVersion + V1_AUDIO_ENABLED);
+
+        deviceModel = deviceInfo.get("ModelName").asText();
+
+        statistics.put("Device Information#Article Number", deviceInfo.get("ArticleNumber").asText());
+        statistics.put("Device Information#Model Name", deviceModel);
+        statistics.put("Device Information#Serial Number", deviceInfo.get("SerialNumber").asText());
+
+        statistics.put("Device Status#Uptime (sec)", deviceInfo.get("CurrentUptime").asText());
+        statistics.put("Device Status#Uptime Total (sec)", deviceInfo.get("TotalUptime").asText());
+        statistics.put("Device Status#First Used", deviceInfo.get("FirstUsed").asText());
+        statistics.put("Device Status#In Use", deviceInfo.get("InUse").asText());
+        statistics.put("Device Status#Status", DEVICE_STATUSES.get(deviceInfo.get("Status").asInt()));
+        statistics.put("Device Status#Sharing", deviceInfo.get("Sharing").asText());
+        addStatisticsProperty(statistics, "Device Status#Status Message", deviceInfo.get("StatusMessage"));
+
+        statistics.put("Device Sensors#Cpu Temperature (C)", deviceInfo.get("Sensors").get("CpuTemperature").asText());
+        statistics.put("Device Sensors#Pcie Temperature (C)", deviceInfo.get("Sensors").get("PcieTemperature").asText());
+        statistics.put("Device Sensors#Sio Temperature (C)", deviceInfo.get("Sensors").get("SioTemperature").asText());
+        addStatisticsProperty(statistics, "Last used", deviceInfo.get("LastUsed"));

```

Comment:
Maybe it better to extract these line to kind of populate method?

Reply:
Sure thing, will do